targets/nrf5x/bluetooth.c

```diff
@@ -2731,18 +2731,43 @@ void jsble_advertising_stop() {
                            NRF_RADIO_NOTIFICATION_DISTANCE_NONE);
    APP_ERROR_CHECK(err_code);
 
+#if PEER_MANAGER_ENABLED
+   peer_manager_init(false /*don't erase_bonds*/);
+   pm_privacy_params_t privacy_params = {
+     .privacy_mode = BLE_GAP_PRIVACY_MODE_OFF,
+     .private_addr_type = BLE_GAP_ADDR_TYPE_RANDOM_PRIVATE_RESOLVABLE,
+     .private_addr_cycle_s = BLE_GAP_DEFAULT_PRIVATE_ADDR_CYCLE_INTERVAL_S,
+     .p_device_irk = NULL
+   };
+   {
+     JsVar *options = jsvObjectGetChildIfExists(execInfo.hiddenRoot, BLE_NAME_PRIVACY);
+     pm_privacy_params_t privacy;
+     if (options && bleVarToPrivacy(options, &privacy)) {
+       privacy_params = privacy;
+     }
+     jsvUnLock(options);
+   }
+   err_code = pm_privacy_set(&privacy_params);
+   if (err_code) jsiConsolePrintf("pm_privacy_set failed: 0x%x\n", err_code);
+#endif
+
 #ifdef NRF52_SERIES
    // Set MAC address
    JsVar *v = jsvObjectGetChildIfExists(execInfo.hiddenRoot, BLE_NAME_MAC_ADDRESS);
    if (v) {
      ble_gap_addr_t p_addr;
      if (bleVarToAddr(v, &p_addr)) {
+#if PEER_MANAGER_ENABLED
+       err_code = pm_id_addr_set(&p_addr);
```

Comment:
This change is here because [the docs](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.sdk5.v15.0.0%2Fgroup__peer__manager.html&anchor=ga780f7978bfd31ded77908e2b33e8d05e) state that, if the peer manager is used, `sd_ble_gap_addr_set` must not be called and `pm_id_addr_set` should be used instead.  
I don't know what might happen if we ignore that, but "must not be called" sounds pretty serious to me.  
Ideally this shouldn't break anything.